pkg/templates/pdb/template.go

```diff
@@ -0,0 +1,101 @@
+)
+
+const (
+	templateKey = "pod-disruption-budget"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Pod Disruption Budget not specified",
+		Key:         templateKey,
+		Description: "Flag DeploymentLike objects having no \"Pod Disruption Budget\" set in case if the replica count is more than the minimum specified value",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.DeploymentLike,
+				objectkinds.PodDisruptionBudget},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				var results []diagnostic.Diagnostic
+
+				replicas, _ := extract.Replicas(object.K8sObject)
+				if int(replicas) > p.MinReplicas {
+					namespaceForDeploymentLike := object.K8sObject.GetNamespace()
+					PDBFound := false
+					maxUnavailable := false
+					labels := extract.Labels(object.K8sObject)
+
+					pdbMaxUnavailable, pdbSelectors := gatherPDBs(lintCtx, namespaceForDeploymentLike)
+					if !(pdbSelectors == nil) {
+						for pdbName, pdbSel := range pdbSelectors {
+							if isMatch(labels, pdbSel) {
+								PDBFound = true
+								if !(pdbMaxUnavailable == nil) && (pdbMaxUnavailable[pdbName] > p.MinimumMaxUnavailableCriteria) {
+									maxUnavailable = true
+								}
+							}
+						}
+
+						if !PDBFound {
+							return []diagnostic.Diagnostic{{Message: fmt.Sprintf("Object has %d replicas, but there is no respective \"Pod Disruption Budget\" match found in the namespace %q.\n", replicas, namespaceForDeploymentLike)}}
+						} else if PDBFound && !maxUnavailable {
+							return []diagnostic.Diagnostic{{Message: fmt.Sprintf("Object has a \"Pod Disruption Budget\" match found in the namespace %q, but no or empty \"maxUnavailable\" field is set in object manifest.\n", namespaceForDeploymentLike)}}
+						}
+					}
+				}
+				return results
+			}, nil
+		}),
+	})
+}
+
+func gatherPDBs(lintCtx lintcontext.LintContext, namespace string) (map[string]int, map[string]map[string]string) {
+	var listOfPDBSelector = make(map[string]map[string]string)
+	var listOfPDBMaxUnavailable = map[string]int{}
+
+	for _, object := range lintCtx.Objects() {
+		pdb, ok := object.K8sObject.(*pdbV1.PodDisruptionBudget)
+		if ok {
+			if pdb.Namespace == namespace {
+				nameOfPDB := pdb.Name
+				PDBSelectors := pdb.Spec.Selector.MatchLabels
+				listOfPDBSelector[nameOfPDB] = PDBSelectors
+				if !(pdb.Spec.MaxUnavailable == nil) {
+					maxUnavailable := (pdb.Spec.MaxUnavailable).IntValue()
+					listOfPDBMaxUnavailable[nameOfPDB] = maxUnavailable
+				}
+			}
+		} else {
+			continue
+		}
+	}
+	return listOfPDBMaxUnavailable, listOfPDBSelector
+}
+
+func isMatch(labels, selectors map[string]string) bool {
+	if len(selectors) > len(labels) {
```

Comment:
Does this have to be true?  You could have more labels on an object than selectors right?